src/Offer/ReadModel/Metadata/OfferMetadataProjector.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Offer\ReadModel\Metadata;
+
+use Broadway\Domain\DomainMessage;
+use Broadway\Domain\Metadata;
+use Broadway\EventHandling\EventListener;
+use CultuurNet\UDB3\EntityNotFoundException;
+use CultuurNet\UDB3\Event\Events\EventCopied;
+use CultuurNet\UDB3\Event\Events\EventCreated;
+use CultuurNet\UDB3\Event\Events\EventImportedFromUDB2;
+use CultuurNet\UDB3\EventHandling\DelegateEventHandlingToSpecificMethodTrait;
+use CultuurNet\UDB3\Place\Events\PlaceCreated;
+use CultuurNet\UDB3\Place\Events\PlaceImportedFromUDB2;
+
+class OfferMetadataProjector implements EventListener
+{
+    use DelegateEventHandlingToSpecificMethodTrait;
+
+    /**
+     * @var OfferMetadataRepository
+     */
+    private $repository;
+
+    /**
+     * @var array<string,string>
+     */
+    private $apiKeyConsumerMapping;
+
+    public function __construct(
+        OfferMetadataRepository $repository,
+        array $apiKeyConsumerMapping
+    ) {
+        $this->repository = $repository;
+        $this->apiKeyConsumerMapping = $apiKeyConsumerMapping;
+    }
+
+    public function applyEventCreated(EventCreated $eventCreated, DomainMessage $domainMessage)
+    {
+        try {
+            $offerMetadata = $this->repository->get($eventCreated->getEventId());
+        } catch (EntityNotFoundException $e) {
+            $offerMetadata = OfferMetadata::default($eventCreated->getEventId());
+        }
+
+        $createdByApiConsumer = $this->getCreatedByApiConsumerFromMetadata($domainMessage->getMetadata());
+        $offerMetadata = $offerMetadata->withCreatedByApiConsumer($createdByApiConsumer);
+
+        $this->repository->save($offerMetadata);
```

Comment:
Maybe here is some duplicated code to extract into a private method?

Reply:
I added this improvement to the follow-up PR: https://github.com/cultuurnet/udb3-silex/pull/636